java/piranha/src/test/resources/com/uber/piranha/XPMethodChainCases.java

```diff
@@ -0,0 +1,94 @@
+package com.uber.piranha;
+
+class XPMethodChainCases {
+
+  interface SomeParam {
+    @BoolParam(key = "other_flag")
+    BoolParameter otherFlag();
+
+    // Annotation and the abstract method should be removed
+    @BoolParam(key = "stale_flag")
+    BoolParameter staleFlag();
+
+    static SomeParam create(Parameter cp) {
+      return null;
+    }
+  }
+
+  public static void foobar(Parameter cp) {
+    SomeParam sp = SomeParam.create(cp);
+    // Matches API
+    if (sp.staleFlag().getValue()) {
+      System.out.println("!");
+    }
+    // Matches API
+    if (!sp.staleFlag().getValue()) {
+      System.out.println("!!!");
+    }
+    SomeParamRev spr = SomeParamRev.create(cp);
+    // Does not match API- is reverse order
+    if (spr.getValue().staleFlag()) {
+      System.out.println("!!!!");
+    }
+    // Does not match API- matches partially
+    if (spr.getValue() != null) {
+      System.out.println("!!!!!");
+    }
+    SomeOtherInterface sot = SomeOtherInterface.create(cp);
+    // Does not match API- matches partially
+    if (sot.staleFlag() != null) {
+      System.out.println("!!");
+    }
+    System.out.println("done!");
+    // Matches API
+    cp.put("", "stale_flag", true);
+    cp.put("", "stale_flag", false);
+    // Do not match API
+    cp.put("", "other_flag", true);
+    cp.put("", "other_flag", false);
+  }
+
+  class TestMethodChainTest {
+    // Matches annotation
+    @PVal(ns = "", key = "stale_flag", val = "true")
+    public void testSomethingTreated() {
+      System.out.println();
+    }
+
+    // Matches annotation
+    @PVal(ns = "", key = "stale_flag", val = "false")
+    public void testSomethingControl() {
+      System.out.println();
+    }
+
+    // Does not match annotation
+    @PVal(ns = "", key = "other_flag", val = "false")
+    public void testSomethingOther() {
+      System.out.println();
+    }
+  }
+  // Mock APIs to create corner case scenarios
```

Comment:
Can these mocks be moved into one of the other files you have created or this needs to be here? 
Since the related code doesn't change, moving it out can reduce the verbosity here.